sklearn/ensemble/gradient_boosting.py

```diff
@@ -1184,6 +1184,37 @@ def _validate_y(self, y):
         # Default implementation
         return y
 
+    def apply(self, X):
+        """Apply trees in the ensemble to X, return leaf indices.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape = [n_samples, n_features]
+            The input samples. Internally, it will be converted to
+            ``dtype=np.float32`` and if a sparse matrix is provided
+            to a sparse ``csr_matrix``.
+
+        Returns
+        -------
+        X_leaves : array_like, shape = [n_samples, n_estimators, n_classes]
```

Comment:
@ogrisel With the latest changes at https://github.com/jmschrei/scikit-learn/commit/4f7f73ed07e69de25a8a1a840896868736a5499d#diff-2229647ab9a84dc25ab3d3d13800cb43R1205 isn't it good enough?